README.md

```diff
@@ -1,29 +1,78 @@
 
-## Notes
+## Example REST Workflow
 
-They've seen rest in the sinatra case, but because HTML/CSS can only really do POST and GET we don't go into it super correctly. A reminder on what REST is and specifically the the routes that are created by default Rails.
+For a real world case study let us pretend that you have a newsletter application, the following is a high level view on how REST works in your app:
 
+1. You fill out the form on the 'New Newsletter' page and click submit.
 
-Most of the web is about exposing data stored in applications.
+2. Information about you, your newsletter content and any additional information such as media items are all sent to the application server.
 
-(feel free to choose a more RESTful example for example domain)
+3. The server interprets the information, recognizes that the request is for a new newsletter and it generates the new record in the database and performs a myriad of background tasks (updates the newsletter counter, possibly sends notification emails, et al)
 
-http verbs mean something urls mean something
+4. Next the server sends a response back to the client, this does not necessarily mean that the newsletter was posted, the response could be that there was an error posting or something like that. However in this case we will say that the newsletter post went through properly, so the server sends a success message and tells the browser what page to go to and render. 
 
-put it all together and we get predictable guessable urls that describe the things they are representing. we call these things a resource.
+5. Lastly the browser receives the server information and gives the user feedback, in this case it shows you a message saying that your newsletter was successfully posted.
 
-/posts vs /pictures
 
-the 7 restful routes 7 restful actions
+## RESTful Conventions in Rails
 
-REST is a concept - not a technology. for right now, just think of it as standard URLS and HTTP verbs that describe what they are trying to do
+Rails has RESTful principles built into its core, so whether you are utilizing the built in view rending system or using the application purely as API you will have the tools necessary to follow standardized routing procedures.
 
-diagram that shows the mirroring of CRUD, to ROutes to controller Actions
+If you were to build out a feature in Rails that entails all of the potential RESTful routes, you would have seven different RESTful URLs, HTTP verbs (more on those later), and actions. As an example, imagine that you have a Newsletter module in your application, if you run ```rake routes``` in the terminal you will receive the following output:
```

Comment:
I would somewhat reframe this starting at something they know (basic CRUD actions) to something new (rake routes and such).

So it would be like

1. If we were trying to create a Newsletter feature (be careful of the word "module" don't want students to think it's a Ruby Module) we would have four main actions. Create, Read, Update and Delete. You know these as the CRUD actions.

2. In addition to the standard CRUD actions, we need an index page to show all of the newsletters

3. And we need a form for creating new newsletters, and a form for updating existing newsletters.

4. All those together and we have seven routes. The GET routes are all routes that are usually render some erb content. The POST and PUT are receivers of forms, and the DELETE is a new type of verb.

5. table of routes as the map from url/verb to english description

6. Thankfully, rails maps these specific things to specific methods or "actions" as they are called in rails. If we had a controller called `NewsletterController` we would define these seven methods and rails automatically will call them based on the correct route. (code snippet of the seven methods, with a comment that is the english description)